### ISSUE TITLE
SHA 9322b35a4084a4cdb71d3d71188686c18aa986ad (current master) will generate a test_project_perf_dll/testonly_lib that does not link

### ISSUE THREAD

**Issue by anporumb** [Azure/c-build-tools#28]
```code
1>------ Build started: Project: test_project_perf_dll, Configuration: Debug x64 ------
1>LINK : error LNK2001: unresolved external symbol some_symbol_for_cppunittest_1
1>D:\repos\c_utils\deps\azure-c-build-tools\test_projects\test_project_perf\Debug\test_project_perf_dll.dll : fatal error LNK1120: 1 unresolved externals
```

this seems to have as source the fact that "some_symbol_for_cppunittest_1" is defined by test_runner_switcher which is not part of "test_project_perf_dll/testonly_lib".

The gate needs to be updated to catch this error.

Best Regards,
Andrei Porumb

**Comment by dcristoloveanu** [comment]
This shall be fixed by #73. In the sense that the test projects will move to C test runner switcher.

**Comment by dcristoloveanu** [comment]
We are now building differently the test projects
